examples/README.md

```diff
@@ -0,0 +1,141 @@
+# How to use the library
+
+This library provide the tools to use together a Redpanda schema registry and a Rabbitmq queue system to 
+consume and publish messages.
+
+# Consumers
+
+A consumer is a process that has the functionality of removing a message from the top of the queue,
+ process it, and go to the next message, doing the same with any subsequent messages until the queue
+is empty. In that moment it stays waiting until new messages are published in the queue, so it can start 
+again.
+
+## How a LabShare consumer process works
+
+A LabShare consumer adds some extra process to the default consumer behaviour. The process of a message using a LabShare consumer can be structured in different stages:
+
+* Message decoding: in this stage the consumer will read the headers attached to the message, and will 
+use them to deserialize the contents of the message, by using the format and compresion codec specified 
```

Comment:
Do you mean compression codec?  I would have described this as

```suggestion
use them to deserialize the contents of the message, by using the schema subject and version number specified 
```

unless you mean

```suggestion
use them to deserialize the contents of the message, by using the format and text encoding specified 
```

Reply:
There are 2 additions to the library: so the text can be 'json' or 'binary' which is specified with the 'encoder_type'; and then there is the compression that could happen when it is 'binary', it could be compressed with either any of the options: 'null', 'deflate' or 'snappy'.  I mean that before validating with the schema, the contents of the message were deserialized from format (json or binary) and compression format (null, deflate, snappy) which I think would happen before parsing with the schema.